### ISSUE TITLE
Default to targeting the maximum supported macOS version

### ISSUE THREAD

**Issue by mcandre** [tpoechtrager/osxcross#300]
Could osxcross default to targeting the maximum supported macOS version? I think that could more closely match the behavior of Xcode. For example, C++17 support for standard features like `std::filesystem::path` should work out of the box.

I recognize that such a change would influence how backwards compatibility works. If a developer wishes to support older macOS versions, perhaps that could be implemented more explicitly, rather than relying on the compiler to default to legacy editions.

**Comment by tpoechtrager** [comment]
What does `uname -a && echo "int main(){}"|xcrun clang -xc -v -` print on your macOS system?

**Comment by HinTak** [comment]
I disagree. If somebody is bothered enough to do cross-compiling, he/she likely wants to support the widest outcome, ie the oldest macOS x version supported by the combination of sdk headers and compiler.

**Comment by IanCollins** [comment]
"If somebody is bothered enough to do cross-compiling, he/she likely wants to support the widest outcome"
Some of us **have** to cross-compile and being restricted to older language versions is a major headache when the code base is cross-platform.
The minimum supported version should be configurable.
